File: Problems/No.232_Implement_Queue_using_Stacks/implement_queue_using_stacks.py
class MyQueue(object):

    def __init__(self):
        self.Input = []
        self.Output = []

    def push(self, x):
        """
        :type x: int
        :rtype: None
        """
        self.Input.append(x)
        

    def pop(self):
        """
        :rtype: int
        """
        self.peek()
        num = self.Output[-1]
        self.Output.pop()
        return num
        

    def peek(self):
        """
        :rtype: int
        """
        if not self.Output:
            while self.Input:
                self.Output.append(self.Input[-1])
                self.Input.pop()
        return self.Output[-1]
        

    def empty(self):
        """
        :rtype: bool
        """
        return not (self.Input or self.Output)


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: Problems/No.232_Implement_Queue_using_Stacks/implement_queue_using_stacks.cpp
class MyQueue {

private:
    std::stack<int> Input;
    std::stack<int> Output;

public:
    
    void push(int x) { Input.push(x); }
    
    int pop() {
        
        peek();
        int num = Output.top();
        Output.pop();
        return num;

    }
    
    int peek() {
        
        if (Output.empty()) {
            while (!Input.empty()) {

                Output.push(Input.top());
                Input.pop();

            }
        }

        return Output.top();

    }
    
    bool empty() { return (Input.empty() && Output.empty()); }

};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */
